tracer/src/Datadog.Tracer.Native/interop.cpp

```diff
@@ -22,51 +23,105 @@ EXTERN_C BOOL STDAPICALLTYPE IsProfilerAttached()
 EXTERN_C VOID STDAPICALLTYPE GetAssemblyAndSymbolsBytes(BYTE** pAssemblyArray, int* assemblySize, BYTE** pSymbolsArray,
                                                         int* symbolsSize)
 {
-    return trace::profiler->GetAssemblyAndSymbolsBytes(pAssemblyArray, assemblySize, pSymbolsArray, symbolsSize);
+    if (trace::profiler == nullptr)
+    {
+        trace::Logger::Error("Error in GetAssemblyAndSymbolsBytes call. Tracer CLR Profiler was not initialized.");
+        return;
+    }
+
+    trace::profiler->GetAssemblyAndSymbolsBytes(pAssemblyArray, assemblySize, pSymbolsArray, symbolsSize);
```

Comment:
I'm no big-city C++ person, but did you mean to remove the "return"? I assume it's fine because the return type is `VOID`. If so, WTH C++?!